ruby-builder.js

```diff
@@ -15,17 +16,40 @@ export function getAvailableVersions(platform, engine) {
 
 export async function install(platform, ruby) {
   const rubyPrefix = await downloadAndExtract(platform, ruby)
+  return rubyPrefix
+}
+
+export async function setupPath(rubyPrefix, ruby) {
+  const rubyPath = path.join(rubyPrefix, 'bin', 'ruby')
+  const gemUserDir = cp.execSync(`${rubyPath} -e "puts Gem.user_dir"`).toString().trim()
 
-  if (platform === 'windows-latest') {
-    require('./windows').setupPath(undefined, rubyPrefix)
+  const originalPath = process.env['PATH'].split(path.delimiter)
+
+  // macOS & Windows have ruby folders in PATH, Ubuntu does not
```

Comment:
It would be more reliable to always remove any path in PATH which matches `/\bruby\b/i`, no matter the platform.

Reply:
It would be nice to share the common part of this code between the two installers to avoid duplication.
I think Windows just needs to add a few extra entries.

Reply:
It looks like there is `/usr/bin/ruby` in PATH though on Ubuntu and macOS, and we shouldn't remove `/usr/bin` from PATH of course.

Reply:
> It would be nice to share the common part

I was debating that also.   I'll see what I can do.

Reply:
> It looks like there is /usr/bin/ruby in PATH though on Ubuntu and macOS

Maybe you're looking at the which output?

See [actions-image-testing](https://github.com/MSP-Greg/actions-image-testing/runs/491413875?check_suite_focus=true#step:4:1)

Reply:
Yes, I was looking at the checks on this PR, e.g. https://github.com/ruby/setup-ruby/pull/32/checks?check_run_id=491284376

Reply:
> Yes, I was looking at the checks on this PR

I guess without a hard link to the line, I don't know.  But the job log outputs `/usr/bin/ruby` as the output of a `which -a ruby` command...

Do you want the two lines of code that determine the `--user-install` bin folder commented out or removed?

Reply:
Removed, maintaining commented code is no fun.